compiler/src/dotty/tools/dotc/core/SymDenotations.scala

```diff
@@ -2246,9 +2246,7 @@ object SymDenotations {
           if (pcls.isCompleting) recur(pobjs1, acc)
           else
             val pobjMembers = pcls.nonPrivateMembersNamed(name).filterWithPredicate { d =>
-              // Drop members of `Any` and `Object`
-              val owner = d.symbol.maybeOwner
-              (owner ne defn.AnyClass) && (owner ne defn.ObjectClass)
+              !defn.topClasses.contains(d.symbol.maybeOwner) // Drop members of top classes
```

Comment:
Previously we did not do it for `AnyValClass`. It seems that we have a constructor there as we do not initialize it with [`ensureCtor = false` in `completeClass`](https://github.com/lampepfl/dotty/pull/10670/files#diff-91bb9b80218c5b1c9fb90d562a52c1aa7a271d795ffffa86c9cd8d1cc133e5ffR265). Not sure if this does a difference in practice.

Reply:
I think it would not make a difference, since objects cannot extend `AnyVal`. Anyway, it's clearer to have the rule work for all top classes.